src/transactions/financial.ts

```diff
@@ -263,3 +309,61 @@ export const constructTransferCreatorCoinTransaction = (
     }
   );
 };
+
+const buildBuyCreatorCoinMetadata = (params: BuyCreatorCoinRequestParams) => {
+  const metadata = new TransactionMetadataCreatorCoin();
+  metadata.profilePublicKey = bs58PublicKeyToCompressedBytes(
+    params.CreatorPublicKeyBase58Check
+  );
+  metadata.operationType = 0;
+  metadata.desoToSellNanos = params.DeSoToSellNanos;
+  metadata.desoToAddNanos = params.DeSoToAddNanos ?? 0;
+  metadata.minCreatorCoinExpectedNanos = params.MinDeSoExpectedNanos ?? 0;
```

Comment:
So if we don't have this value, we will want to set it to a max uint64 to estimate an upper bound on the size of the transaction. Also it should be params.MinCreatorCoinExpectedNanos instead of params.MinDeSoExpectedNanos

The main reason we can't actually support local construction for CC buys is it's hard to estimate how many CCs a user should expect for their buy because the CC curve math is hard to replicate in JS.